boards/arc/nsim/doc/index.rst

```diff
@@ -82,56 +118,207 @@ nsim, and display the following console output:
+   nSIM). However if you are able to launch application on multi-core nsim target that means you can
+   simply :ref:`debug with MDB debugger <board_arc_nsim_debugging_mwdt>`.
+   It's the nSIM with ARC GDB restriction, real HW multi-core ARC targets can be debugged with ARC
+   GDB.
+
+.. note::
+   Currently we don't support debug with GDB on `HAPS`_ platform.
+
+.. note::
+   The normal ``west debug`` command won't work for debugging applications using nsim boards
+   because both the nSIM simulator and the debugger (either GDB or MDB) use the same console for
+   input / output.
+   In case of GDB debugger you can use separate terminal windows for GDB and nSIM to avoid
+   intermixing their output. For the MDB debugger you can simply use GUI bode.
+
+After building your application, open two terminal windows. In terminal one, use nSIM to start a GDB
+server and wait for a remote connection with following command:
+
+.. code-block:: console
+
+   west debugserver --runner arc-nsim
+
+In terminal two, connect to the GDB server using ARC GDB. You can find it in Zephyr SDK:
+
+* for the ARCv2 targets you should use :file:`arc-zephyr-elf-gdb`
+* for the ARCv3 targets you should use :file:`arc64-zephyr-elf-gdb`
+
+This command loads the symbol table from the elf binary file, for example the
+:file:`build/zephyr/zephyr.elf` file:
+
+.. code-block:: console
+
+   arc-zephyr-elf-gdb  -ex 'target remote localhost:3333' -ex load build/zephyr/zephyr.elf
 
 Now the debug environment has been set up, you can debug the application with gdb commands.
 
+Modifying the configuration
+***************************
+
+If you've decided to modify existing nsim configuration or even create a new one it's required
+to maintain alignment between
+
+* Zephyr OS configuration
+* nSIM & MDB configuration
+* GNU & MWDT toolchain compiler options
+
+.. note::
+   The ``.tcf`` configuration files are not supported by Zephyr directly. There are multiple
+   reasons for that. ``.tcf`` perfectly suits when we building bare-metal single-thread application -
+   in that case we want to pass all the compiler options from ``.tcf`` to compiler, so all the HW
+   features will be used by application and we get optimal code.
+   Ihe simulation is completely different when we are talking about building the multi-thread
+   features-reach operation system. Of course we still can simply build all code with all the
+   options from ``.tcf`` - but that may be far from optimal solution. For example, such approach
+   require so save & restore full register context for all tasks (and sometimes even for
+   interrupts). And for DSP-enabled or for FPU-enabled systems that leads to dozens of extra
+   registers save and restore even if the most of the user and kernel tasks don't actually use
+   DSP or FPU. Instead we prefer to fine-tune the HW features usage which (with all its pros)
+   require us to maintain the separate from ``.tcf`` configuration.
+
```

Comment:
I understand tcf-based flows are not supported, but the rationale can be argued. TCF allows for adding and overriding items in the TCF file on the command line if required.

Reply:
> rationale can be argued

That's true. However I'd say it's better to provide at least some rationale and describe the current state (that tcf is not supported directly)